docs/technical/journey-flow.md

```diff
@@ -0,0 +1,224 @@
+# Journey Flow
+
+- [Overview](#overview)
+- [Journey Map configuration](#journey-map-configuration)
+    - [Mapping](#mapping)
+    - [File structure](#file-structure)
+    - [Branching](#branching)
+
+## Overview
+
+Part of this POC is to make it easier to visualise, create and maintain journeys through the 
+pages within the client facing front end.
+
+This can be achieved by placing reusable journeys into modules each containing a set of pages or
+routes that when combined make a self contained reusable journey.
+Examples of such modules would be an address module, a contact module and a file upload module.
+
+A way of configuring the journey within the module and connections between modules is with mapping files.
+Within the POC, [YAML](https://yaml.org/start.html) was used to describe the journey configuration within
+each mapping file.
+
+## Journey Map configuration
+
+The best way to describe this is with an example:
+
+### Mapping
+Please note that each of the entries within the following files ultimately generate routes within a hapi service.
+
+Example mapping files for a simple journey:
+- Root map:
+```yaml
+--- # Root map 
+
+home:
+  path: "/"
+  route: home.route
+  next: applicant
+
+applicant:
+  path: "/applicant"
+  module: contact
+  next: complete
+
+complete:
+  path: "/complete"
+  route: complete.route
+```
+- Contact map:
+```yaml
+--- # Contact map
+
+name:
+  path: "/name"
+  route: contact-name.route
+  next: address
+
+address:
+  path: "/address"
+  module: address
+  next: email
+
+email:
+  path: "/email"
+  route: contact-email.route
+  next: return
+``` 
+- Address map:
+```yaml
+--- # Address map
+
+search:
+  path: "/search"
+  route: address-search.route
+  next: select
+  
+select:
+  path: "/select"
+  route: address-select.route
+  next: entry
+
+entry:
+  path: "/entry"
+  route: address-entry.route
+  next: return
+```
+
+The idea is that the navigation through the routes (pages) starts in the root map and flows
+between the routes via the next property.  When the module property is set, the flow moves
+to the start of that modules map and flows through that map.  When a next property within that map is set
+to return, the flow returns to the previous map and continues.
+
+As I have included no [branching](#branching) in the above map, I would expect the paths (pages) to be traversed in the following order:
+```text
+- /
+- /applicant/name
+- /applicant/address/search
+- /applicant/address/select
+- /applicant/address/select
```

Comment:
This looks like a typo that should be `/applicant/address/entry`

Reply:
Fixed the typo, will now merge